import AXIOS from '../config/axios.config';
import { headers, URL } from './constants';

export const Teacher = {

  getTeacherList() {
    return AXIOS.get(`${URL.TEACHER}`, {
      headers: headers(),
    })
      .then(response => {
        return response.data;
      })
      .catch(error => {
        if (error && error.response && error.response && error.response.data)
          return Promise.reject(error.response.data.message);
        return Promise.reject('Ups! Al parecer hay un error desconocido.');
      });
  },
  findTeacherById(id) {
    return AXIOS.get(`${URL.TEACHER}/${id}`, {
      headers: headers(),
    })
      .then(response => {
        return response.data;
      })
      .catch(error => {
        if (error && error.response && error.response && error.response.data)
          return Promise.reject(error.response.data.message);
        return Promise.reject('Ups! Al parecer hay un error desconocido.');
      });
  },
  update(teacher) {
    return AXIOS.patch(`${URL.TEACHER}/${teacher.id}`, teacher, {
      headers: headers(),
    })
      .then(response => {
        return response.data;
      })
      .catch(error => {
        if (error && error.response && error.response && error.response.data)
          return Promise.reject(error.response.data.message);
        return Promise.reject('Ups! Al parecer hay un error desconocido.');
      });
  },

  saveTeacher(teacher) {
    return AXIOS.post(`${URL.TEACHER}`, teacher, {
      headers: headers(),
    })
      .then(response => {
        console.log(response);
        return response.data;
      })
      .catch(error => {
        if (error && error.response && error.response && error.response.data)
          return Promise.reject(error.response.data.message);
        return Promise.reject('Ups! Al parecer hay un error desconocido.');
      });
  },
  delete(teacherId) {
    return AXIOS.delete(`${URL.TEACHER}/${teacherId}`, {
      headers: headers(),
    })
      .then(response => {
        return response.data;
      })
      .catch(error => {
        if (error && error.response && error.response && error.response.data)
          return Promise.reject(error.response.data.message);
        return Promise.reject('Ups! Al parecer hay un error desconocido.');
      });
  },
};

export default Teacher;
